pkg/network/pod/annotations/generator.go

```diff
@@ -109,12 +107,7 @@ func (g Generator) GenerateFromActivePod(vmi *v1.VirtualMachineInstance, pod *k8
 		return iface.SRIOV != nil || vmispec.HasBindingPluginDeviceInfo(iface, g.clusterConfig.GetNetworkBindings())
 	})
 
-	networkStatusAnnotation := pod.Annotations[networkv1.NetworkStatusAnnot]
-	networkDeviceInfoMap, err := deviceinfo.MapNetworkNameToDeviceInfo(vmi.Spec.Networks, networkStatusAnnotation, ifaces)
-	if err != nil {
-		log.Log.Warningf("failed to create network device-info-map: %v", err)
-	}
-
```

Comment:
Prior to this change, `MapNetworkNameToDeviceInfo` could have returned an error in one of the following cases:
1. The value of the network-status annotation was empty.
2. The unmarshaling of the raw JSON string to `[]networkv1.NetworkStatus` had failed for some reason.

By making `MapNetworkNameToDeviceInfo` take a `[]networkv1.NetworkStatus` as an input parameter - the two possible error paths are no longer relevant - thus there is no need for this warning IMO.

Please note that `multus.NetworkStatusesFromPod` has a log message in case the JSON unmarshaling fails:
https://github.com/kubevirt/kubevirt/blob/635ba09c13144b3deecd1114c88ec50bc9d65dd4/pkg/network/multus/status.go#L45-L55

Reply:
Thanks for explaining, I think we good.